### PR TITLE
Make data migrations idempotent

### PR DESCRIPTION
**Pull request by mattwr18**

### CODE REVIEW

File: db/data/20240722092527_add_organization_id_to_requests.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 class AddOrganizationIdToRequests < ActiveRecord::Migration[6.1]
   def up
     ActiveRecord::Base.transaction do
-      organization = Organization.singleton
+      organization = Organization.first
```
Comment (soey) [inline db/data/20240722092527_add_organization_id_to_requests.rb:6]: what happens later on instances when there are multiple organizations? I would even add sth like return if Organization.count != 1 to ensure that it is only run with exactly one organization

File: db/data/20240722092527_add_organization_id_to_requests.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@ def up
 
   def down
     ActiveRecord::Base.transaction do
-      organization = Organization.singleton
+      organization = Organization.first
       return unless organization
```
Comment (soey) [inline db/data/20240722092527_add_organization_id_to_requests.rb:20]: see above

File: db/data/20240726061113_add_organization_id_to_activity_notifications.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@
 class AddOrganizationIdToActivityNotifications < ActiveRecord::Migration[6.1]
   def up
     ActiveRecord::Base.transaction do
-      organization = Organization.singleton
+      organization = Organization.first
       return unless organization
```
Comment (soey) [inline db/data/20240726061113_add_organization_id_to_activity_notifications.rb:7]: see above

File: db/data/20240726061113_add_organization_id_to_activity_notifications.rb [hunk 4]
```diff
@@ -16,7 +16,7 @@ def up
 
   def down
     ActiveRecord::Base.transaction do
-      organization = Organization.singleton
+      organization = Organization.first
       return unless organization
```
Comment (soey) [inline db/data/20240726061113_add_organization_id_to_activity_notifications.rb:20]: see above

File: db/data/20240726063319_add_organization_id_to_taggings.rb [hunk 5]
```diff
@@ -3,7 +3,7 @@
 class AddOrganizationIdToTaggings < ActiveRecord::Migration[6.1]
   def up
     ActiveRecord::Base.transaction do
-      organization = Organization.singleton
+      organization = Organization.first
       return unless organization
```
Comment (soey) [inline db/data/20240726063319_add_organization_id_to_taggings.rb:7]: see above

File: db/data/20240726063319_add_organization_id_to_taggings.rb [hunk 6]
```diff
@@ -16,7 +16,7 @@ def up
 
   def down
     ActiveRecord::Base.transaction do
-      organization = Organization.singleton
+      organization = Organization.first
       return unless organization
```
Comment (soey) [inline db/data/20240726063319_add_organization_id_to_taggings.rb:20]: see above

**Review by soey — COMMENTED** [review]
Ouch ..yes .. data migrations and code .. they are still painful. I would be even more pessimistic and add safeguards also for the cases if we have multiple organizations

**Comment by mattwr18** [comment]
> Ouch ..yes .. data migrations and code .. they are still painful. I would be even more pessimistic and add safeguards also for the cases if we have multiple organizations

this is a good point and makes me want to remove this data migrate gem and go back to the temp rake tasks/ansible plays, which can be run once on whichever instances they need to be run and aren't run automatically on deploy.

**Comment by mattwr18** [comment]
Superseded by https://github.com/tactilenews/100eyes/pull/2042

**Pull request closed without merging**